playground/src/index.js

```diff
@@ -33,8 +40,73 @@ import '@wordpress/block-library/build-style/theme.css';
 import '@wordpress/format-library/build-style/style.css';
 /* eslint-enable no-restricted-syntax */
 
+const fauxEntitySuggestions = [
+	{
+		id: uniqueId(),
+		title: 'Hello Page',
+		type: 'Page',
+		url: '/hello-page/',
+	},
+	{
+		id: uniqueId(),
+		title: 'Hello Post',
+		type: 'Post',
+		url: '/hello-post/',
+	},
+	{
+		id: uniqueId(),
+		title: 'Hello Another One',
+		type: 'Page',
+		url: '/hello-another-one/',
+	},
+	{
+		id: uniqueId(),
+		title: 'This is another Post with a much longer title just to be really annoying and to try and break the UI',
+		type: 'Post',
+		url: '/this-is-another-post-with-a-much-longer-title-just-to-be-really-annoying-and-to-try-and-break-the-ui/',
+	},
+];
+
 function App() {
 	const [ blocks, updateBlocks ] = useState( [] );
+	const [ link, setLink ] = useState( fauxEntitySuggestions[ 0 ] );
+	const [ linkSettings, setLinkSettings ] = useState( {
+		'new-tab': false,
+	} );
+
+	const [ isVisible, setIsVisible ] = useState( true );
+
+	/* eslint-disable @wordpress/react-no-unsafe-timeout */
+	const timeout = ( ms ) => {
+		return new Promise( ( resolve ) => setTimeout( resolve, ms ) );
+	};
+	/* eslint-enable @wordpress/react-no-unsafe-timeout */
+
+	const fetchFauxEntitySuggestions = async () => {
+		// Simulate network
+		await timeout( random( 200, 1000 ) );
+
+		return fauxEntitySuggestions;
+	};
+
+	const handleOnKeyDownEvent = ( event, suggestion ) => {
+		console.log( `onKeyDown - Key code: ${ event.keyCode }` );
+		if ( null !== suggestion ) {
+			console.log( `suggestion: ${ suggestion }` );
+		}
+
+		// Do not stop propagation for ESCAPE key
+		if ( ESCAPE === event.keyCode ) {
+			return;
+		}
+
+		event.stopPropagation();
+	};
+
+	const handleOnKeyPressEvent = ( event) => {
```

Comment:
```suggestion
	const handleOnKeyPressEvent = ( event ) => {
```

Reply:
Hmm, I guess these playground changes need to be removed from the PR? Maybe at that point `fetchLinkSuggestions` can also be removed from `URLInput`, as I think it's only in use here.

Reply:
> I guess these playground changes need to be removed from the PR

Absolutely they do! 

> at that point fetchLinkSuggestions can also be removed from URLInput, as I think it's only in use here.

Damn. I'm actually making heavy use of that in my tests in order to be able to control the data returned when you perform a search. Is there an easier way to mock out [this call](https://github.com/WordPress/gutenberg/blob/56e730be6bf2583f58845b69215450131766bb55/packages/block-editor/src/components/url-input/index.js#L320)?

Reply:
Oh, just leave it in that case, but make sure it's `experimental`.

Reply:
Done.